paper-linear-progress.html

```diff
@@ -0,0 +1,916 @@
+          } else {
+            this.__needsBufferingAnimationOnAttached = true;
+          }
+        } else {
+          if (isAnimationReady) {
+            this.cancelAnimation();
+          } else {
+            this.__needsBufferingAnimationOnAttached = false;
+          }
+        }
+      },
+
+      _closedChanged: function(isClosed) {
+        this.toggleClass('closed', isClosed);
+      },
+
+      _applyBufferingDotsBackground: function() {
+        var bgColor = window.getComputedStyle(this.$.buffer).backgroundColor;
+        var prologue = 'url("data:image/svg+xml;charset=UTF-8,';
+        // NOTE(traviskaufman): Single-quotes are needed around SVG attributes in order for it to
+        // render properly in all browsers, hence the escaped single-quote strings as opposed to
+        // double-quoting.
+        var bufferingDotsSVG = '' +
+          '<svg version=\'1.1\' ' +
+              'xmlns=\'http://www.w3.org/2000/svg\' ' +
+              'xmlns:xlink=\'http://www.w3.org/1999/xlink\' ' +
+              'x=\'0px\' y=\'0px\' ' +
+              'enable-background=\'new 0 0 5 2\' ' +
+              'xml:space=\'preserve\' ' +
+              'viewBox=\'0 0 5 2\' ' +
+              'preserveAspectRatio=\'none slice\'> ' +
+            '<circle cx=\'1\' cy=\'1\' r=\'1\' fill=\'' + bgColor + '\'/>' +
+          '</svg>';
+        var epilogue = '")';
+        var encodedDots = encodeURIComponent(bufferingDotsSVG).replace(/'/g, '%27');
+        var svgBackgroundUrl = prologue + encodedDots + epilogue;
+        this.$.bufferingDots.style.backgroundImage = svgBackgroundUrl;
+      },
+
+      _applyAnimationConfig: function() {
+        var computedHeight = parseInt(window.getComputedStyle(this).height, 10);
+        var bufferingDots = this.$.bufferingDots;
+        // Since the buffering dots SVG has a viewBox that's 5px by 2px, to translate the dots
+        // by the width of the viewbox (and therefore show the illusion of a "continuous" moving
+        // element), we need to translate it by 2.5 * the height of the element.
+        var translationAmt = 2.5 * computedHeight;
+        var timing = {
+          duration: 250,
+          iterations: Infinity,
+          easing: 'linear'
+        };
+
+        this.animationConfig = {
+          'buffering': {
+            name: 'transform-animation',
+            node: bufferingDots,
+            transformFrom: 'translateX(0px)',
+            // Use -1 to move the dots left.
+            transformTo: 'translateX(' + (translationAmt * -1) + 'px)',
+            timing: timing
+          },
+
+          'buffering-reverse': {
+            name: 'transform-animation',
+            node: bufferingDots,
+            transformFrom: 'translateX(0px)',
+            transformTo: 'translateX(' + translationAmt + 'px)',
+            timing: timing
+          }
+        };
+      },
+
+      _forceLayoutIfHasClass: function(className) {
+        var classList = this.classList;
```

Comment:
This is a minor hygiene detail, but it would probably be better if we used `Polymer.dom(this).classList` here.